### PR TITLE
Fix (#545) Ports are now properly merged.

### PR DESCRIPTION
**Pull request by iocanel**
Fixes: #545 

This pull request fixes some issues with the a bean merger, that is used to merge configuration together (property, annotation and framework configurration).

The main issue, is that when merging arrays, lists and set we didn't take ihertitence into account. This PR fixes that and also improves on the way we counfugre ports.

### CODE REVIEW

File: core/src/main/java/io/dekorate/kubernetes/configurator/ApplyContainerPort.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    this(containerPort, Arrays.asList(names));
+  }
+
+  @Override
+  public void visit(BaseConfigFluent<?> config) {
+    Predicate<PortBuilder> predicate = p -> names.contains(p.getName());
+    if (config.hasMatchingPort(predicate)) {
+      config.editMatchingPort(predicate).withContainerPort(containerPort).endPort();
+    } else {
+      String name = names.get(0);
```
Comment (geoand) [inline core/src/main/java/io/dekorate/kubernetes/configurator/ApplyContainerPort.java:47]: Are we sure this always exists?

File: core/src/main/java/io/dekorate/utils/Beans.java [hunk 2]
```diff
@@ -167,8 +166,9 @@
     try {
       Class originClass = origin.getClass().getComponentType();
       List result = new ArrayList<>();
-      List originList = Arrays.asList((Object[])origin);
-      List overrideList = Arrays.asList((Object[])override);
+      //We wrap the Arrays.asList() in an ArrayList so that we can mutate it.
```
Comment (geoand) [inline core/src/main/java/io/dekorate/utils/Beans.java:169]: Inception :P

**Comment by geoand** [comment]
Would be nice to see a test that shows that #545 is indeed addressed. Is that possible?

File: core/src/main/java/io/dekorate/kubernetes/configurator/ApplyContainerPort.java (hunk 1 above)
Comment (iocanel) [reply core/src/main/java/io/dekorate/kubernetes/configurator/ApplyContainerPort.java:47]: As this is only used in one place, yes we are sure.
It's a bad idea nonetheless. 
Fix incoming.

**Comment by iocanel** [comment]
> Would be nice to see a test that shows that #545 is indeed addressed. Is that possible?

Yeap!

**Comment by iocanel** [comment]
Added test!

**Pull request merged**